momo_modules/gsm_module/src/gsm_serial.c

```diff
@@ -27,6 +26,7 @@ void enable_serial()
 	SPBRGH = 0;
 	SPBRGL = 16;
 
+	CREN = 0;
```

Comment:
Naw, I would just put a comment in explaining why its there and then 
leave as it.

On 1/7/15 1:53 PM, Austin McGee wrote:
>
> In momo_modules/gsm_module/src/gsm_serial.c 
> <https://github.com/WellDone/MoMo-Firmware/pull/154#discussion-diff-22619336>:
>
> > @@ -27,6 +26,7 @@ void enable_serial()
> >  	SPBRGH = 0;
> >  	SPBRGL = 16;
> >
> > +	CREN = 0;
>
> This was introduced to catch an edge-case error that could occur if 
> the gsm module was turned off and causes a serial OERR. In that case, 
> to clear the error you would disable and then re-enable CREN (as is 
> done when reading or writing chars). The alternative (and possibly 
> better option) would be to disable serial communication when turning 
> off the SIM900 (currently it stays "on"), but doing it in the init 
> method as well ensures that there's no way an error could be present. 
> Does that make sense? We should possibly do both and guard the |CREN = 
> 0;| line with an |if (OERR == 1)| condition to make it obvious what's 
> going on.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/WellDone/MoMo-Firmware/pull/154/files#r22619336>.
>